spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/StreamCompletionProvider.java

```diff
@@ -35,6 +39,15 @@
 	@Autowired
 	private List<? extends ExpansionStrategy> completionExpansionStrategies = new ArrayList<>();
 
+	@PostConstruct
+	public void sortStrategies() {
+		// Here we mean HIGHEST_PRECEDENCE to win over others, so must be *last*
+		AnnotationAwareOrderComparator.sort(completionExpansionStrategies);
```

Comment:
Maybe. I guess '-<tab>' and '--<tab>' could be handled by the same
completer. Will look into it (DSL parser may fail differently)
Le 19 déc. 2015 3:53 PM, "Mark Fisher" <notifications@github.com> a écrit :

> In
> spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/StreamCompletionProvider.java
> <https://github.com/spring-cloud/spring-cloud-dataflow/pull/263#discussion_r48093176>
> :
>
> > @@ -35,6 +39,15 @@
> >  	@Autowired
> >  	private List<? extends ExpansionStrategy> completionExpansionStrategies = new ArrayList<>();
> >
> > +	@PostConstruct
> > +	public void sortStrategies() {
> > +		// Here we mean HIGHEST_PRECEDENCE to win over others, so must be *last*
> > +		AnnotationAwareOrderComparator.sort(completionExpansionStrategies);
>
> *circular dependency* sounds like another red flag... any way to refactor
> to avoid it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spring-cloud/spring-cloud-dataflow/pull/263/files#r48093176>
> .
>